packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedFieldMessageEmbed.ts

```diff
@@ -0,0 +1,167 @@
+	 * import { MessageEmbed } from 'discord.js';
+	 *
+	 * new PaginationFieldMessageEmbed().setTemplate(new MessageEmbed().setTitle('Test pager embed')).make().run(message.author, message.channel);
+	 * ```
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 * import { MessageEmbed } from 'discord.js';
+	 *
+	 * new PaginationFieldMessageEmbed().setTemplate({ title: 'Test pager embed' }).make().run(message.author, message.channel);
+	 * ```
+	 */
+	public setTemplate(template: MessageEmbedOptions | MessageEmbed) {
+		this.embedTemplate = template instanceof MessageEmbed ? template : new MessageEmbed(template);
+		return this;
+	}
+
+	/**
+	 * The way in which the items of the object array will be represented in the embed field to be paginated.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginationFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'Discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 * @param value Item information
+	 */
+	public formatField(value: (value: T, index: number, array: T[]) => any) {
+		this.items = this.items.map(value);
+		return this;
+	}
+
+	/**
+	 * Build the item pagination of the configured array.
+	 * It is required to put the `.make()` and `.run()` method at the end in order to start.
+	 *
+	 * @example
+	 * ```typescript
+	 * import { PaginationFieldMessageEmbed } from '@sapphire/discord.js-utilities';
+	 *
+	 * new PaginationFieldMessageEmbed()
+	 *    .setTitleField('Test field')
+	 *    .setTemplate({ embed })
+	 *    .setItems([
+	 *       { title: 'Sapphire Framework', value: 'discord.js Framework' },
+	 *       { title: 'Sapphire Framework 2', value: 'discord.js Framework 2' },
+	 *       { title: 'Sapphire Framework 3', value: 'Discord.js Framework 3' }
+	 *     ])
+	 *    .formatField((item) => `${item.title}\n${item.value}`)
+	 *    .make()
+	 *    .run(message);
+	 * ```
+	 */
+	public make() {
+		if (!this.fieldTitle.length) throw new Error('The title of the field to format must have a value.');
+		if (!this.items.length) throw new Error('The array introduced has no items.');
+		if (!this.items[0]) throw new Error('The format of the array items is incorrect.');
+
+		this.totalPages = Math.ceil(this.items.length / this.itemsPerPage);
+		this.generatePages();
+		return this;
+	}
+
+	private generatePages() {
+		const parsedTemplate = this.embedTemplate instanceof MessageEmbed ? (this.embedTemplate.toJSON() as MessageEmbedOptions) : this.embedTemplate;
```

Comment:
The naming of this variable is unclear - it misleadingly implies that the template was parsed in some way, whereas actually, it contains the template as an embed options object.

Reply:
When I say that it was analyzed, I mean that it was analyzed whether the template was already instantiated by the MessageEmbed class or it is the object as such.

Reply:
Parsing is to break something down into syntactic components, by definition. This has not been parsed - it has been converted to an object.

Reply:
Check the new variable name.

Reply:
This is resolved now.